backend/Origam.Workbench.Services/ServiceAgentFactory.cs

```diff
@@ -69,7 +69,7 @@ public List<IServiceAgent> XslFunctionProviderServiceAgents
                             {
                                 if (log.IsDebugEnabled)
                                 {
-                                    log.DebugFormat("Can't load service {0}, skipping {1}", service.Name,
+                                    log.DebugFormat("Can't load service {0}, skipping {1}", service?.Name,
```

Comment:
The `service` shall never be null, although it technically could be. Isn't there a way how to ensure it is never null so we do not have to check here but also anywhere else?

Reply:
The whole pull request was created in the spirit of "let's add some checks inside the logging if statements so that the logging never fails". So if it is a problem to add the check here why is it ok every wehere else? I mean there are probably other checks in places where a problem is unlikely but they were added anyway in case somebody makes a mistake somewhere else (like in the serviceItemProvider.ChildItemsByType method) in which case at least the logging will work.